File: backend/src/main/java/com/equisoft/dca/backend/user/service/UserService.java
package com.equisoft.dca.backend.user.service;

import java.util.List;
import java.util.Optional;

import javax.validation.Valid;

import org.springframework.validation.annotation.Validated;

import com.equisoft.dca.backend.common.OnCreate;
import com.equisoft.dca.backend.common.OnUpdate;
import com.equisoft.dca.backend.user.model.User;

public interface UserService {
	Optional<User> findById(int id);

	Optional<User> findByIdentifier(String identifier);

	@Validated(OnCreate.class)
	User save(String currentUserIdentifier, @Valid User user);

	@Validated(OnUpdate.class)
	User update(String currentUserIdentifier, @Valid User user);

	void deleteById(String currentUserIdentifier, Integer id);

	List<User> findAll();

	List<User> findByOrganizationId(int organizationId);
}

File: react/src/lib/services/UserService.ts
import User from '../domain/entities/User';
import UserRepository from '../domain/repositories/UserRepository';

export default class UserService {
    constructor(private userRepository: UserRepository) {
    }

    async add(user: User): Promise<User> {
        return this.userRepository.add(user);
    }

    async update(user: User): Promise<void> {
        return this.userRepository.update(user);
    }

    async delete(id: string): Promise<void> {
        return this.userRepository.delete(id);
    }

    async findAll(): Promise<User[]> {
        return this.userRepository.findAll();
    }

    async findById(id: string): Promise<User> {
        return this.userRepository.findById(id);
    }
}